ccl/changefeedccl: TestParallelIOMetrics failed

ccl/changefeedccl.TestParallelIOMetrics [failed](https://tanzanite.cluster.engflow.com/invocations/default/a059d5cb-3a18-462b-a813-b12093701c87?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-24.2.5-rc @ [ee625f92aaca9a68ff53706626b2c5050a91a6f4](https://github.com/cockroachdb/cockroach/commits/ee625f92aaca9a68ff53706626b2c5050a91a6f4):


```
=== RUN   TestParallelIOMetrics
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestParallelIOMetrics1714763469
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:9370: -- test log scope end --
test logs left over in: outputs.zip/logTestParallelIOMetrics1714763469
--- FAIL: TestParallelIOMetrics (238.95s)
=== RUN   TestParallelIOMetrics/pubsub
    helpers_test.go:872: making server as system tenant
    helpers_test.go:948: making pubsub feed factory
    changefeed_test.go:9342: condition failed to evaluate within 3m45s: from changefeed_test.go:9345: waiting for pending keys: 0
    --- FAIL: TestParallelIOMetrics/pubsub (238.94s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133335 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.1 release-blocker]
- #130390 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-24.2.2-rc release-blocker]
- #129917 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-24.2 release-blocker]
- #125295 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-master]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParallelIOMetrics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43778

ccl/changefeedccl.TestParallelIOMetrics [failed](https://tanzanite.cluster.engflow.com/invocations/default/0f7599a6-b8a6-4643-b65e-3a18189c2a57?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-24.2.5-rc @ [2e9644dd47ffff1befda9f8f6051d85e16bf4182](https://github.com/cockroachdb/cockroach/commits/2e9644dd47ffff1befda9f8f6051d85e16bf4182):


```
=== RUN   TestParallelIOMetrics
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestParallelIOMetrics3336053241
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:9370: -- test log scope end --
test logs left over in: outputs.zip/logTestParallelIOMetrics3336053241
--- FAIL: TestParallelIOMetrics (153.10s)
=== RUN   TestParallelIOMetrics/pubsub
    helpers_test.go:869: making server as secondary tenant
    helpers_test.go:948: making pubsub feed factory
    changefeed_test.go:9366: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:9366
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1120
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1155
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParallelIOMetrics/pubsub
    --- FAIL: TestParallelIOMetrics/pubsub (153.10s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134188 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.3]
- #133335 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.1 release-blocker]
- #130390 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-24.2.2-rc release-blocker]
- #129917 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-24.2 release-blocker]
- #125295 ccl/changefeedccl: TestParallelIOMetrics failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParallelIOMetrics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -9317,13 +9317,13 @@ func TestParallelIOMetrics(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	// Add delay so queuing occurs, which results in the below metrics being
+	// nonzero.
+	defer testingEnableQueuingDelay()()
+
 	testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
 		registry := s.Server.JobRegistry().(*jobs.Registry)
 		metrics := registry.MetricsStruct().Changefeed.(*Metrics).AggMetrics
-
-		// Add delay so queuing occurs, which results in the below metrics being
-		// nonzero.
-		defer testingEnableQueuingDelay()()
 
 		db := sqlutils.MakeSQLRunner(s.DB)
 		db.Exec(t, `SET CLUSTER SETTING changefeed.new_pubsub_sink_enabled = true`)

